bin/ch-test

```diff
@@ -186,18 +337,35 @@ fi
 
 # Ensure BATS_TMPDIR is set to /tmp (issue #278).
 if [[ -n $BATS_TMPDIR ]]; then
-    if [[ ! $BATS_TMPDIR == '/tmp/'* ]]; then
+    if [[ "$BATS_TMPDIR" != '/tmp'* ]]; then
         fatal "BATS_TMPDIR: must be /tmp but found $BATS_TMPDIR (issue #278)"
     fi
 fi
 
+# Create a directory in user space to hold auto-generated test artifacts
+tmpdir=${HOME}/.charliecloud/ch-test.tmp.$USER
+if [[ ! -d $tmpdir ]]; then
+    mkdir -p "$tmpdir"
+    chmod 700 "$tmpdir"
+    chmod 700 "$(dirname "$tmpdir")"
```

Comment:
(1) Don't mess with `$tmpdir`'s parent.
(2) Don't use `-p`, since you know it doesn't exist.
(3) I think the `mkdir`/`chmod` combination sets up a race condition and maybe other issues. Use `-m 700` instead.